index.bs

```diff
@@ -1523,6 +1652,51 @@ To get the <dfn>handle for an object</dfn> given |realm|, |ownership type| and
 
 </div>
 
+<div algorithm>
+To get the <dfn>shared id for an object</dfn> given a |realm| and |object|:
+
+1. Let |object| be [=unwrapped=] |object|.
+
+1. If |object| is not a [=platform object=], return <code>null</code>.
+
+1. If |object| is an [=/Element=]:
+
+  1. Let |browsing context| be the result of [=get the browsing context=] of the
+     |realm|.
+
+  1. If |browsing context| is <code>null</code>, return <code>null</code>.
+
+  1. Return the result of [=trying=] to [=get or create a web element reference=]
+     with |object| and |browsing context|.
+
+1. If |object| is a [=/ShadowRoot=]:
+
+  1. Let |browsing context| be the result of [=get the browsing context=] of the
+     |realm|.
+
+  1. If |browsing context| is <code>null</code>, return <code>null</code>.
+
+  1. Return the result of [=trying=] to [=get or create a shadow root reference=]
+     with |object| and |browsing context|.
+
+1. Let |window| be the result of [=get the window=] of the |realm|.
```

Comment:
Before this, should we handle the case that _object_ is a window object, using the WebDriver classic window handle format?